### PR TITLE
Added avg_pool2d function for TensorFlow frontend (#1548)

### PR DESCRIPTION
**Pull request by Subramaniam-dot**
avg_pool2d #22783

- Implemented avg_pool2d in nn.py to enhance neural network functionalities.
- Addresses issue #22783 in the ongoing effort to expand TensorFlow frontend capabilities.

### CODE REVIEW

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hey! Seems like your PR is missing the code addition. Can you please push the changes and then request a review? Thanks!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
